app/src/test/java/com/android/solvit/shared/model/request/ServiceRequestRepositoryFirebaseTest.kt

```diff
@@ -157,18 +161,23 @@ class ServiceRequestRepositoryFirebaseTest {
   @Test
   fun documentToServiceRequest_validDocumentSnapshot_returnsServiceRequest() {
     // Set up mock DocumentSnapshot
+    val time = Timestamp.now()
 
     `when`(mockDocumentSnapshot.id).thenReturn("1")
     `when`(mockDocumentSnapshot.getString("title")).thenReturn("Test Request")
     `when`(mockDocumentSnapshot.getString("description")).thenReturn("Test Description")
     `when`(mockDocumentSnapshot.getString("userId")).thenReturn("1")
-    `when`(mockDocumentSnapshot.getTimestamp("dueDate")).thenReturn(Timestamp.now())
+    `when`(mockDocumentSnapshot.getTimestamp("dueDate")).thenReturn(time)
     `when`(mockDocumentSnapshot.getDouble("location.latitude")).thenReturn(0.0)
     `when`(mockDocumentSnapshot.getDouble("location.longitude")).thenReturn(0.0)
     `when`(mockDocumentSnapshot.getString("location.name")).thenReturn("EPFL")
     `when`(mockDocumentSnapshot.getString("imageUrl")).thenReturn(null)
     `when`(mockDocumentSnapshot.getString("type")).thenReturn("CLEANER")
     `when`(mockDocumentSnapshot.getString("status")).thenReturn("PENDING")
+    `when`(mockDocumentSnapshot.getString("providerId")).thenReturn("1")
```

Comment:
Same as for the first android test ! Using local emulator could make your testing process easier !